src/Models/ShallowWaterModels/update_shallow_water_state.jl

```diff
@@ -10,11 +9,6 @@ Fill halo regions for `model.solution` and `model.tracers`.
 """
 function update_state!(model::ShallowWaterModel)
 
-    # Mask immersed fields
-    masking_events = Tuple(mask_immersed_field!(field) for field in model.solution)
-
-    wait(device(model.architecture), MultiEvent(masking_events))
-
```

Comment:
Do we need to mask velocities for the shallow water model? I think we might.

Reply:
The above lines don't touch tracers (just transport and height) so it might be fine to keep that.

Reply:
You're saying I should put those lines back on?

Reply:
Yes, I think we need to